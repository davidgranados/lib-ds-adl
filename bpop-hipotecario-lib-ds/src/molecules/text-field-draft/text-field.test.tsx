import React from "react";
import { mount, ReactWrapper } from "enzyme";
import TextField from "./text-field";

describe("<TextField />", () => {
  let wrapper: ReactWrapper<typeof TextField>;
  const props = {
    id: "test-text-field-id",
    label: "test text field label",
    value: "test value",
    caption: "test caption",
    placeholder: "test placeholder",
    size: "md",
    onFocus: jest.fn(),
    onBlur: jest.fn(),
    onChange: jest.fn(),
    onInput: jest.fn(),
  };
  describe("<TextField /> with valid value", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof TextField>(<TextField {...props} />);
    });
    it("should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
    describe("<Label />", () => {
      it("should render correctly", () => {
        // when
        const labelElement = wrapper.find("label");
        // then
        expect(labelElement.prop("htmlFor")).toEqual(props.id);
        expect(labelElement.prop("children")).toEqual(props.label);
      });
    });
    describe("<InputBase />", () => {
      it("should render correctly", () => {
        // when
        const inputElement = wrapper.find("input");
        // then
        expect(inputElement.prop("id")).toEqual(props.id);
        expect(inputElement.prop("value")).toEqual(props.value);
        expect(inputElement.prop("type")).toEqual("text");
        expect(inputElement.prop("placeholder")).toEqual(props.placeholder);
        expect(inputElement.hasClass(`size-${props.size}`)).toBe(true);
        expect(inputElement.hasClass(`filled`)).toBe(true);
      });
    });
  });
  describe("<TextField /> with error caption", () => {
    beforeEach(() => {
      jest.clearAllMocks();
      // given
      wrapper = mount<typeof TextField>(<TextField {...props} status={"error"} />);
    });
    it("should match snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
    describe("<InputBase />", () => {
      it("should have error class", () => {
        // when
        const inputElement = wrapper.find("input");
        // then
        expect(inputElement.hasClass("error")).toBe(true);
      });
    });
    describe("<Typography />", () => {
      it("should render correctly", () => {
        // when
        const labelElement = wrapper.find("span");
        // then
        expect(labelElement.prop("children")).toEqual(props.caption);
      });
    });
  });
});
