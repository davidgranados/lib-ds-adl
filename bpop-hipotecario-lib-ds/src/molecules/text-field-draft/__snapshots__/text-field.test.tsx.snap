// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> <TextField /> with error caption should match snapshot 1`] = `
<div
  className="wrapper"
>
  <label
    className="default"
    htmlFor="test-text-field-id"
  >
    test text field label
  </label>
  <input
    className="default size-md error filled"
    id="test-text-field-id"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    onInput={[Function]}
    placeholder="test placeholder"
    type="text"
    value="test value"
  />
  <span
    className="default body-sm weight-regular error-600 line-height-body-sm"
  >
    test caption
  </span>
</div>
`;

exports[`<TextField /> <TextField /> with valid value should match snapshot 1`] = `
<div
  className="wrapper"
>
  <label
    className="default"
    htmlFor="test-text-field-id"
  >
    test text field label
  </label>
  <input
    className="default size-md filled"
    id="test-text-field-id"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    onInput={[Function]}
    placeholder="test placeholder"
    type="text"
    value="test value"
  />
</div>
`;
